target/scripts/helpers/postfix.sh

```diff
@@ -92,39 +92,42 @@ function _vhost_ldap_support() {
 # /etc/aliases is handled by `alias.sh` and uses `postalias` to update the Postfix alias database. No need for `postmap`.
 # http://www.postfix.org/postalias.1.html
 
-# Add an key with an value to Postfix's main configuration file
+# Add a key with a value to Postfix's main configuration file
 # or update an existing key. An already existing key can be updated
 # by either appending to the existing value (default) or by prepending.
 #
 # @param ${1} = key name in Postfix's main configuration file
 # @param ${2} = new value (appended or prepended)
-# @param ${3} = "append" (default) or "prepend" [OPTIONAL]
+# @param ${3} = action "append" (default) or "prepend" [OPTIONAL]
 function _add_to_or_update_postfix_main() {
   local KEY=${1:?Key name is required}
   local NEW_VALUE=${2:?New value is required}
   local ACTION=${3:-append}
+  local CURRENT_VALUE
 
-  # If entry does not exist, add it - otherwise update with ACTION:
-  if ! grep -q -E "^${KEY}" /etc/postfix/main.cf; then
+  # Get current value from /etc/postfix/main.cf
+  CURRENT_VALUE=$(postconf -h "${KEY}" 2>/dev/null)
```

Comment:
You'll probably want to call this helper first, as reading from `main.cf` frequently can compound the stalling effect otherwise, slowing down startup time:

https://github.com/docker-mailserver/docker-mailserver/blob/c8a0bfd36195281b629916d64edf0a2c1fbd89d1/target/scripts/helpers/utils.sh#L57-L66

I brought this up early in [the original PR review for this helper](https://github.com/docker-mailserver/docker-mailserver/pull/3484#pullrequestreview-1581477237) also thinking `postconf` would be a better way to go about it.

Reply:
Done [81b7785](https://github.com/docker-mailserver/docker-mailserver/pull/3503/commits/81b7785ff5182c379f48a57b63d712e3b2f22fdb)